src/lib.rs

```diff
@@ -126,12 +131,22 @@ impl<L: LogLevel> Verbosity<L> {
         self.log_level().is_none()
     }
 
-    fn verbosity(&self) -> i8 {
-        level_value(L::default()) - (self.quiet as i8) + (self.verbose as i8)
+    fn verbosity(&self) -> i16 {
+        level_value(L::default()) - (self.quiet as i16) + (self.verbose as i16)
```

Comment:
How bad would it be if we instead switched this to
```suggestion
        level_value(L::default()).saturating_add((self.verbose - self.quiet)))
```

Reply:
In general, people shouldn't be providing so many that they saturate

Reply:
A single `-q` would underflow (0u8 - 1u8). This would work fine in debug mode, and fail in release mode.

While in general that is indeed true that users calling the executable should not in general write paramaters that would cause it to crash, similarly apps should be robust in the face of users doing strange things. I don't think it's safe for a library to crash with an overflow given bad inputs like this. It may be better to clamp the total number of parameters parsed to some given number.

Given Clap is used to process command lines, that some commands will be part of administrative scripts running with elevated privileges, being able to control a crash in a command with specific parameters lead to potential (very low impact) CVEs.

It's possible that switching this internally to u8 would be a better idea, and make the OFF condition return 0 instead of -1. This would allow for u8 and saturating to 0 throughout (except for when calculating the delta between quote and verbosity).

I agree it makes sense to pull this change out as separate commit. I think it probably makes sense as a new PR for review purposes, or is the idea of this split to pull out this change strictly for commit history / message purposes?

Reply:
Switched to u8 instead in #115